components/pedigree/resources/src/main/resources/pedigree/pedigreeDate.js

```diff
@@ -200,11 +209,15 @@ define([], function(){
         },
 
         // Returns an integer or null.
-        // Iff "failsafe" returns first year of the decade if year is not set and decade is
-        getYear: function(failsafe) {
+        // Iff "failsafe" returns a value even if only decade is availabe (and year is not):
+        //  - iff "average" the middle year of the decade is returned, otherwise the first year of the decade.
+        getYear: function(failsafe, average) {
             if (this.isComplete() && this.year == null && failsafe) {
                 // remove trailing "s" from the decade && convert to integer
                 var year = parseInt( this.decade.slice(0,-1) );
+                if (average) {
+                    year += 5;
```

Comment:
Good point, I forgot we have those extremely fuzzy dates.

It will fail (work incorrectly) by returning 1705. Not sure 1800-s and 1700-s matter for BOADICEA, but to fix this better we probably need different suffixes for decades and millenniums. Though after available date range has been extended down to 1850 this problem becomes purely academic IMHO.

Do you want me to fix this?